### ISSUE TITLE
Signal : impossible to define the "range" facet value as a variable

### ISSUE THREAD

**Issue by mazarsju** [gama-platform/gama#1305]
In the following code, impossible to replace ```range:10``` by ```range:rangeValue``` :

```
model MyFirstModel

global {
	int gridsize <- 10;
	geometry shape <- rectangle(10#m,10#m);
	float decayValue<-0.1;
	float proportionValue <- 5.0;
	float variationValue <- 0.0;
	float rangeValue <- 10;
	int timeVar <- 0;

	init {
		create signalSpecie number:1 with: (location: {5.0,5.0});
	}
}


grid cells height: gridsize width: gridsize neighbours:8 use_regular_agents: false use_individual_shapes: false {
	const neighbours type: list of: cells <- self neighbors_at 1;

	rgb color <- #black update:(sign<0.001)? #black : rgb(0.0,sign,sign);
	
	reflex updateTime{
		if (timeVar=20){
			timeVar <- 0;
		}
		else {
			timeVar <- timeVar+1;
		}
	}
} 

species signalSpecie skills:[moving] {
	signal sign update:(timeVar=0) ? 5*240 : 0 decay:decayValue environment:cells range:/*rangeValue*/10 propagation:diffusion proportion:proportionValue variation:variationValue;
}


experiment Signal type: gui {
	output {
		display a type: opengl {
			grid cells;
		}
	}
}
```

**Comment by AlexisDrogoul** [comment]
What is the error message ? Actually, I think that signals are expecting a constant range (as it allows to cache  the neighbors of cells), but it should be indicated as a warning (saying that the initial value of rangeValue will be taken into account), not as an error.

**Comment by mazarsju** [comment]
java.lang.NullPointerException
	at msi.gaml.variables.SignalVariable.<init>(SignalVariable.java:171)
	at gaml.additions.GamlAdditions$9.create(GamlAdditions.java:93)
	at msi.gaml.descriptions.SymbolProto.create(SymbolProto.java:190)
	at msi.gaml.descriptions.SymbolDescription.compile(SymbolDescription.java:575)
	at msi.gaml.descriptions.SymbolDescription.compileChildren(SymbolDescription.java:599)
	at msi.gaml.descriptions.SymbolDescription.compile(SymbolDescription.java:584)
	at msi.gaml.descriptions.SymbolDescription.compileChildren(SymbolDescription.java:599)
	at msi.gaml.descriptions.SymbolDescription.compile(SymbolDescription.java:584)
	at msi.gama.lang.gaml.resource.GamlResource.build(GamlResource.java:395)
	at msi.gama.lang.gaml.resource.GamlModelBuilder.compile(GamlModelBuilder.java:110)
	at msi.gama.lang.gaml.resource.GamlModelBuilder.compile(GamlModelBuilder.java:91)
	at msi.gama.lang.gaml.resource.GamlModelBuilder.compile(GamlModelBuilder.java:86)
	at msi.gama.lang.gaml.ui.editor.GamlEditor$1$1.exec(GamlEditor.java:277)
	at msi.gama.lang.gaml.ui.editor.GamlEditor$1$1.exec(GamlEditor.java:1)
	at org.eclipse.xtext.resource.OutdatedStateManager.exec(OutdatedStateManager.java:121)
	at org.eclipse.xtext.ui.editor.model.XtextDocument$XtextDocumentLocker.internalReadOnly(XtextDocument.java:520)
	at org.eclipse.xtext.ui.editor.model.XtextDocument$XtextDocumentLocker.readOnly(XtextDocument.java:492)
	at org.eclipse.xtext.ui.editor.model.XtextDocument.readOnly(XtextDocument.java:133)
	at msi.gama.lang.gaml.ui.editor.GamlEditor$1.widgetSelected(GamlEditor.java:272)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:248)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4353)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1061)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1085)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1070)
	at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:782)
	at msi.gama.gui.swt.controls.FlatButton.doButtonClicked(FlatButton.java:109)
	at msi.gama.gui.swt.controls.FlatButton.handleEvent(FlatButton.java:77)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4353)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1061)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4172)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3761)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$9.run(PartRenderingEngine.java:1151)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1032)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:148)
	at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:636)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:579)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:150)
	at msi.gama.gui.swt.Application.start(Application.java:125)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:380)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:235)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:648)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:603)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1465)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1438)
Storing the metadata just created (or recreated) while reading it for L/my_new_project/models/new.gaml
Writing back metadata to L/my_new_project/models/new.gaml
Success: sync info written
Finally: updating the decorator manager

**Comment by AlexisDrogoul** [comment]
OK. This seems to have been fixed as well by commit 483f674146cc0c5f0a3726026c87008a57660ca9 . Please verify.
